src/compiler/emitter.ts

```diff
@@ -1015,7 +1015,7 @@ namespace ts {
         function emitLiteral(node: LiteralLikeNode) {
             const text = getLiteralTextOfNode(node);
             if ((printerOptions.sourceMap || printerOptions.inlineSourceMap)
-                && (node.kind === SyntaxKind.StringLiteral || isTemplateLiteralKind(node.kind))) {
+                && (node.kind === SyntaxKind.StringLiteral || isTemplateLiteralToken(node))) {

```

Comment:
the kind function is supposedly faster, since it is mono-morphic, where as the one with node is polymorphic.

Reply:
the function itself will not be inlined if it is polymorphic, that is why we have some functions, like in the binder, that take the kind.. the kind is always a number, and thus not polymorphic.